doc/zmq.md

```diff
@@ -0,0 +1,155 @@
+
+In this fashion, external software can use a trusted Bitcoin Core
+daemon to do the heavy lifting of communicating with the P2P network,
+while still receiving network information asynchronously in real
+time. As transactions and blocks arrive via the P2P network, Bitcoin
+Core will apply all the validation and standardization rules to this
+data, only passing along through ZeroMQ those items that pass.
+
+ZeroMQ sockets are self-connecting and self-healing; that is, connects
+made between two endpoints will be automatically restored after an
+outage, and either end may be freely started or stopped in any order.
+
+Because ZeroMQ is message oriented, subscribers receive transactions
+and blocks all-at-once and do not need to implement any sort of
+buffering or reassembly.
+
+## Prerequisites
+
+The ZeroMQ feature in Bitcoin Core uses only a very small part of the
+ZeroMQ C API, and is thus compatible with any version of ZeroMQ
+from 2.1 onward, including all versions in the 3.x and 4.x release
+series.  Typically, it is packaged by distributions as something like
+*libzmq-dev*.
+
+The C++ wrapper for ZeroMQ is *not* needed.
+
+## Enabling
+
+By default, the ZeroMQ port functionality is enabled. Two steps are
+required to enable--compiling in the ZeroMQ code, and configuring
+runtime operation on the command-line or configuration file.
+
+    $ ./configure --enable-zmq (other options)
+
+This will produce a binary that is capable of providing the ZeroMQ
+facility, but will not do so until also configured properly.
+
+## Configuration
+
+Currently, the following notifications are supported:
+
+    -zmqpubhashtx=address
+    -zmqpubhashblock=address
+    -zmqpubrawblock=address
+    -zmqpubrawtx=address
+
+The socket type is PUB and the address must be a valid ZeroMQ
+socket address. The same address can be used in more than one notification.
+
+For instance:
+
+    $ bitcoind -zmqpubhashtx=tcp://127.0.0.1:28332 -zmqpubrawtx=ipc:///tmp/bitcoind.tx.raw
+
+Each PUB notification has a topic and body, where the header
+corresponds to the notification type. For instance, for the notification
+`-zmqpubhashtx` the topic is `hashtx`.
+
+These options can also be provided in bitcoin.conf.
+
+ZeroMQ endpoint specifiers for TCP (and others) are documented in the
+[ZeroMQ API](http://api.zeromq.org).
+
+## Operation
+
+ZeroMQ publish sockets prepend each data item with an arbitrary topic
+prefix that allows subscriber clients to request only those items with
+a matching prefix.  When publishing, bitcoind will prepend the topic
+"TXN" (no quotes, no null terminator) to the binary, serialized form
+of a published transaction, and "BLK" to the binary, serialized form
+of a published block.
+
+Client side, then, the ZeroMQ subscriber socket must have the
+ZMQ_SUBSCRIBE option set to one or either of these prefixes; without
+doing so will result in no messages arriving.
+
+Here is a small example, in the Python language, using the
+*python-zmq* wrapper:
```

Comment:
IMO better mention `contrib/zmq/zmq_sub.py` here. Otherwise we have to maintain two python examples.